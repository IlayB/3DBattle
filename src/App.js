import RetroBackground from './components/retroBackground';

const App = () => {
  return (
    <div className="App">
      <RetroBackground />
    </div>
  );
};

export default App;
