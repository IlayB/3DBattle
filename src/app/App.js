import { BrowserRouter } from 'react-router-dom';
import { getToken } from 'firebase/messaging';

import { createTheme, ThemeProvider } from '@mui/material';

import RetroBackground from '../components/retroBackground';
import TopMenuNavigation from '../components/topMenuNavigation/TopMenuNavigation';
import { AppRoutes } from './route/AppRoutes';
import { messaging } from '../auth/firebaseApp';
import { useEffect } from 'react';

const MuiTheme = createTheme({
  typography: {
    fontFamily: 'New Rocker',
    allVariants: {
      color: 'white',
    },
    button: {
      textTransform: 'none',
    },
  },
  palette: {
    mode: 'dark',
  },
});

const App = () => {
  useEffect(() => {
    getToken(messaging, {
      vapidKey:
        'BPWRKU0cZTpZlMMK71z57nS-Bubi4NknfuIFC5NYpXVaPBfiPdA2BBOpyshiLqop-t4Em_amhl5X3vFBtI6dEvE',
    })
      .then((currentToken) => {
        if (currentToken) {
          console.log('TOKEN', currentToken);
        } else {
          console.log('No registration token available. Request permission to generate one.');
        }
      })
      .catch((err) => {
        console.log('An error occurred while retrieving token. ', err);
      });
  }, []);

  useEffect(() => {
    navigator.serviceWorker.addEventListener('message', ({ data }) =>
      console.log('GOT MESSAGE', data),
    );
  }, []);
  return (
    <div className="App">
      <ThemeProvider theme={MuiTheme}>
        <RetroBackground />
        <BrowserRouter basename={process.env.PUBLIC_URL}>
          <TopMenuNavigation />
          <AppRoutes />
        </BrowserRouter>
      </ThemeProvider>
    </div>
  );
};

export default App;
