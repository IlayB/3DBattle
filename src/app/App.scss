.App {
  margin: 0;
  height: 100%;
  background-color: red;
}