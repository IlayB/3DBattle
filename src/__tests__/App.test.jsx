import { render } from '@testing-library/react';

import App from '../app/App';

describe('App', () => {
  it('renders correctly', () => {
    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
  });
});
