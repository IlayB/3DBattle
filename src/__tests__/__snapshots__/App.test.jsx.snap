// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <div
        id="retrobg"
      >
        <div
          id="retrobg-sky"
        >
          <div
            id="retrobg-stars"
          >
            <div
              class="retrobg-star"
              style="left: 5%; top: 55%; transform: scale( 2 );"
            />
            <div
              class="retrobg-star"
              style="left: 7%; top: 5%; transform: scale( 2 );"
            />
            <div
              class="retrobg-star"
              style="left: 10%; top: 45%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 12%; top: 35%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 15%; top: 39%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 20%; top: 10%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 35%; top: 50%; transform: scale( 2 );"
            />
            <div
              class="retrobg-star"
              style="left: 40%; top: 16%; transform: scale( 2 );"
            />
            <div
              class="retrobg-star"
              style="left: 43%; top: 28%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 45%; top: 30%; transform: scale( 3 );"
            />
            <div
              class="retrobg-star"
              style="left: 55%; top: 18%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 60%; top: 23%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 62%; top: 44%; transform: scale( 2 );"
            />
            <div
              class="retrobg-star"
              style="left: 67%; top: 27%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 75%; top: 10%; transform: scale( 2 );"
            />
            <div
              class="retrobg-star"
              style="left: 80%; top: 25%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 83%; top: 57%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 90%; top: 29%; transform: scale( 2 );"
            />
            <div
              class="retrobg-star"
              style="left: 95%; top: 5%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 96%; top: 72%; transform: scale( 1 );"
            />
            <div
              class="retrobg-star"
              style="left: 98%; top: 70%; transform: scale( 3 );"
            />
          </div>
          <div
            id="retrobg-sunWrap"
          >
            <div
              id="retrobg-sun"
            />
          </div>
          <div
            id="retrobg-mountains"
          >
            <div
              class="retrobg-mountain"
              id="retrobg-mountains-left"
            />
            <div
              class="retrobg-mountain"
              id="retrobg-mountains-right"
            />
          </div>
          <div
            id="retrobg-cityWrap"
          >
            <div
              id="retrobg-city"
            >
              <div
                class="retrobg-building"
                style="left: 4.0%; height: 20%; width: 3.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 6.0%; height: 50%; width: 1.5%;"
              />
              <div
                class="retrobg-building"
                style="left: 8.0%; height: 25%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 12.0%; height: 30%; width: 3.0%;"
              />
              <div
                class="retrobg-building retrobg-antenna"
                style="left: 13.0%; height: 55%; width: 3.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 17.0%; height: 20%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 18.5%; height: 70%; width: 1.5%;"
              />
              <div
                class="retrobg-building"
                style="left: 20.0%; height: 30%; width: 4.0%;"
              />
              <div
                class="retrobg-building retrobg-antenna"
                style="left: 21.5%; height: 80%; width: 2.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 25.0%; height: 60%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 28.0%; height: 40%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 30.0%; height: 70%; width: 4.0%;"
              />
              <div
                class="retrobg-building retrobg-antenna"
                style="left: 35.0%; height: 65%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 38.0%; height: 40%; width: 3.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 42.0%; height: 60%; width: 2.0%;"
              />
              <div
                class="retrobg-building retrobg-antenna"
                style="left: 43.0%; height: 85%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 45.0%; height: 40%; width: 3.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 48.0%; height: 25%; width: 3.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 50.0%; height: 80%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 52.0%; height: 32%; width: 5.0%;"
              />
              <div
                class="retrobg-building retrobg-antenna"
                style="left: 55.0%; height: 55%; width: 3.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 58.0%; height: 45%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 61.0%; height: 90%; width: 4.0%;"
              />
              <div
                class="retrobg-building retrobg-antenna"
                style="left: 66.0%; height: 99%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 69.0%; height: 30%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 73.5%; height: 90%; width: 2.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 72.0%; height: 70%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 75.0%; height: 60%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 80.0%; height: 40%; width: 4.0%;"
              />
              <div
                class="retrobg-building retrobg-antenna"
                style="left: 83.0%; height: 70%; width: 4.0%;"
              />
              <div
                class="retrobg-building retrobg-antenna"
                style="left: 87.0%; height: 60%; width: 3.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 93.0%; height: 50%; width: 3.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 91.0%; height: 30%; width: 4.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 94.0%; height: 20%; width: 3.0%;"
              />
              <div
                class="retrobg-building"
                style="left: 98.0%; height: 35%; width: 2.0%;"
              />
            </div>
          </div>
        </div>
        <div
          id="retrobg-ground"
        >
          <div
            id="retrobg-linesWrap"
          >
            <div
              id="retrobg-lines"
            >
              <div
                id="retrobg-vlines"
              >
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
                <div
                  class="retrobg-vline"
                />
              </div>
              <div
                id="retrobg-hlines"
              >
                <div
                  class="retrobg-hline"
                />
                <div
                  class="retrobg-hline"
                />
                <div
                  class="retrobg-hline"
                />
                <div
                  class="retrobg-hline"
                />
                <div
                  class="retrobg-hline"
                />
                <div
                  class="retrobg-hline"
                />
                <div
                  class="retrobg-hline"
                />
                <div
                  class="retrobg-hline"
                />
              </div>
            </div>
          </div>
          <div
            id="retrobg-groundShadow"
          />
        </div>
      </div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1k0i2s9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ne52jz-MuiButtonBase-root-MuiButton-root"
              display="row"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1145ie3-MuiSvgIcon-root"
                data-testid="WhatshotIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
                />
              </svg>
              <h5
                class="MuiTypography-root MuiTypography-h5 css-qlz4m-MuiTypography-root"
              >
                3D Battle
              </h5>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiBox-root css-1m04nb5"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-v52rcf-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1juz4d3-MuiButtonBase-root-MuiButton-root"
              display="row"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1145ie3-MuiSvgIcon-root"
                data-testid="WhatshotIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
                />
              </svg>
              <h5
                class="MuiTypography-root MuiTypography-h5 css-qlz4m-MuiTypography-root"
              >
                3D Battle
              </h5>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiBox-root css-1t6c9ts"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-2nv7f2-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-jb1ygz-MuiTypography-root"
                >
                  Inventory
                </p>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-2nv7f2-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-jb1ygz-MuiTypography-root"
                >
                  Store
                </p>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiBox-root css-2uchni"
            >
              <button
                aria-label="Open settings"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-ontyxu-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    alt="Remy Sharp"
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="/static/images/avatar/2.jpg"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </header>
      <div
        class="MuiBox-root css-ulxmrl"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-jb1ygz-MuiTypography-root"
        >
          Home
        </p>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="MuiPopover-root MuiMenu-root MuiModal-root MuiModal-hidden css-14ogsoa-MuiModal-root-MuiPopover-root-MuiMenu-root"
    id="menu-appbar"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 0; visibility: hidden;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="-1"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiPaper-root MuiMenu-paper MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper css-1if8r2c-MuiPaper-root-MuiMenu-paper-MuiPaper-root-MuiPopover-paper"
      style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
        role="menu"
        tabindex="-1"
      >
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-pwsekp-MuiButtonBase-root-MuiMenuItem-root"
          role="menuitem"
          tabindex="0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gql1ui-MuiTypography-root"
          >
            Profile
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="-1"
    />
  </div>
</body>
`;
